apiserver/facades/controller/crosscontroller/register.go

```diff
@@ -26,7 +27,7 @@ func newStateCrossControllerAPI(ctx facade.Context) (*CrossControllerAPI, error)
 	return NewCrossControllerAPI(
 		ctx.Resources(),
 		func() ([]string, string, error) {
-			return common.StateControllerInfo(st)
+			return controllerInfo(st)
```

Comment:
We should only use the client api addresses for cross controller cmr. For single controller cmr, the agent api addresses are preferable. We don't want traffic within the cloud unnecessarily being forced to use  public (non cloud local) addresses.

Reply:
This is the cross-controller API.